record_test.go

```diff
@@ -254,3 +256,50 @@ func TestRecordBatchDecoding(t *testing.T) {
 		}
 	}
 }
+
+func TestRecordBatchLargeNumRecords(t *testing.T) {
+	numOfRecords := 10 + (2 * math.MaxUint16)
```

Comment:
Not sure how useful making this parameterizable is, when the CRC will change if the value changes, but then I like that it’s apparent and inarguable what length we’re encoding.

Reply:
Yea I only set it as a var to make it obvious what it is. I kept forgetting what I had it set to during testing, so it's probably useful just to have it easily readable like you said haha.